Makefile

```diff
@@ -201,7 +202,10 @@ destroy_%:
 		echo "cloud=$* region=$$region cluster_id=$$index"; \
 		cd "infra/$*"; \
 		terraform workspace select $*-$$index-$$region; \
-		terraform destroy ${terraform_destroy_args} -var-file="../../terraform.tfvars.json" -var=$*_k8s_region=$$region -var=cluster_id=$$index; \
+		cluster_name=`terraform output cluster_name | jq . -r`; \
+		terraform state rm "module.external_dns.helm_release.external_dns[0]"; \
+		terraform state rm "module.external_dns.module.external_dns_helm.helm_release.this[0]"; \
+		terraform destroy ${terraform_destroy_args} -var-file="../../terraform.tfvars.json" -var=$*_k8s_region=$$region -var=cluster_id=$$index -var=cluster_name=$$cluster_name; \
```

Comment:
Can you explain why the external-dns needs to be removed manually in this step?

Reply:
I will move the external-dns to the separate execution runtime, like we have argocd and other addons, otherwise it is a chicken and egg problem, where the k8s cluster can be destroyed prior helm chart deletion.